app/models/daos/slick/UserDAOSlick.scala

```diff
@@ -191,6 +192,94 @@ object UserDAOSlick {
+//
+//    distinctUsers.size.run
+//  }
+
+
+  /**
+   * Returns a count of all users who completed at least one audit task or at least one validation given
+   * the following parameters
+   *
+   * @param timeInterval: can be "today", "yesterday", "week", or "month". If anything else, defaults to "all time"
+   * @param taskCompletedOnly: if true, only counts users who have completed one audit task or at least one validation.
+   *                           Defaults to false.
+   * @param highQualityOnly: if true, only counts users who are marked as high quality. Defaults to false.
+   * @return
+   */
+  def countAllUsersContributed(timeInterval: String = "all time", taskCompletedOnly: Boolean = false, highQualityOnly: Boolean = false): Int = db.withSession { implicit session =>
+
+    // Build up SQL string related to validation and audit task time intervals
+    // Defaults to *not* specifying a time (which is the same thing as "all time")
+    var lblValidationTimeIntervalSql = ""
+    var auditTaskTimeIntervalSql = ""
+    if(timeInterval.equalsIgnoreCase("today")){
+      lblValidationTimeIntervalSql = """WHERE (label_validation.end_timestamp AT TIME ZONE 'PST')::date = (NOW() AT TIME ZONE 'PST')::date"""
+      auditTaskTimeIntervalSql = """AND (audit_task.task_end AT TIME ZONE 'PST')::date = (NOW() AT TIME ZONE 'PST')::date"""
+    }else if(timeInterval.equalsIgnoreCase("yesterday")){
+      lblValidationTimeIntervalSql = """WHERE (label_validation.end_timestamp AT TIME ZONE 'PST')::date = (now() AT TIME ZONE 'PST')::date - interval '1' day"""
+      auditTaskTimeIntervalSql = """AND (audit_task.task_end AT TIME ZONE 'PST')::date = (now() AT TIME ZONE 'PST')::date - interval '1' day"""
+    }else if(timeInterval.equalsIgnoreCase("week")){
+      lblValidationTimeIntervalSql = """WHERE (label_validation.end_timestamp AT TIME ZONE 'PST')::date > DATE_SUB(NOW() AT TIME ZONE 'PST', INTERVAL 1 WEEK)"""
+      auditTaskTimeIntervalSql = """AND (audit_task.task_end AT TIME ZONE 'PST')::date > DATE_SUB(NOW() AT TIME ZONE 'PST', INTERVAL 1 WEEK)"""
+    }else if(timeInterval.equalsIgnoreCase("month")){
+      lblValidationTimeIntervalSql = """WHERE (label_validation.end_timestamp AT TIME ZONE 'PST')::date > DATE_SUB(NOW() AT TIME ZONE 'PST', INTERVAL 1 MONTH)"""
+      auditTaskTimeIntervalSql = """AND (audit_task.task_end AT TIME ZONE 'PST')::date > DATE_SUB(NOW() AT TIME ZONE 'PST', INTERVAL 1 MONTH)"""
+    }
+
+    // Add in the optional SQL WHERE statement for filtering on high quality users
+    var highQualityOnlySql = ""
+    if(highQualityOnly){
+      highQualityOnlySql = """WHERE user_stat.high_quality_manual = TRUE OR user_stat.high_quality_manual IS NULL;"""
+    }
+
+    // Add in the task completion logic. Note: to keep the Scala code clean, if taskCompletedOnly is false
+    // then I put in a SQL statement to check both for all completed and all incompleted tasks.
+    var auditTaskCompletedSql = ""
```

Comment:
You can also just do
```
val auditTaskCompletedSql = if (taskCompletedOnly) {
  "blah"
} else {
  "other blah"
}
```
to avoid mutable variables.